packages/expect/src/__tests__/customEqualityTesters.test.ts

```diff
@@ -0,0 +1,241 @@
+      toIterator([special2, special1]),
+    );
+    expect({a: special1, b: undefined}).not.toStrictEqual({
+      a: special2,
+      b: undefined,
+    });
+    expect({a: 1, b: {c: special1}}).not.toMatchObject({
+      a: 1,
+      b: {c: special2},
+    });
+  });
+
+  it('asymmetric matchers do not pass different special objects', () => {
+    expect([special1]).not.toEqual(expect.arrayContaining([special2]));
+    expect({a: 1, b: {c: special1}}).not.toEqual(
+      expect.objectContaining({b: {c: special2}}),
+    );
+  });
+
+  it('spy matchers do not pass different special objects', () => {
+    const mockFn: (...args: Array<any>) => any = jest.fn(() => specialReturn1);
+    mockFn(...testArgs);
+
+    expect(mockFn).not.toHaveBeenCalledWith(...expectedArgs);
+    expect(mockFn).not.toHaveBeenLastCalledWith(...expectedArgs);
+    expect(mockFn).not.toHaveBeenNthCalledWith(1, ...expectedArgs);
+
+    expect(mockFn).not.toHaveReturnedWith(specialReturn2);
+    expect(mockFn).not.toHaveLastReturnedWith(specialReturn2);
+    expect(mockFn).not.toHaveNthReturnedWith(1, specialReturn2);
+  });
+
+  it('custom matcher does not pass different special objects', () => {
+    expect(special1).not.toSpecialObjectEqual(special2);
+  });
+});
+
+describe('with custom equality testers', () => {
+  let originalTesters: Array<Tester>;
+
+  beforeAll(() => {
+    originalTesters = jestExpect.customEqualityTesters;
+    jestExpect.customEqualityTesters = [...originalTesters, specialObjTester];
+  });
+
+  afterAll(() => {
+    jestExpect.customEqualityTesters = originalTesters;
+  });
+
+  it('basic matchers customTesters do not apply to still do not pass different special objects', () => {
+    expect(special1).not.toBe(special2);
+    expect([special1]).not.toContain(special2);
+  });
+
+  it('basic matchers pass different special objects', () => {
+    expect(special1).toEqual(special1);
+    expect(special1).toEqual(special2);
+    expect([special1, special2]).toEqual([special2, special1]);
+    expect(new Map([['key', special1]])).toEqual(new Map([['key', special2]]));
+    expect(new Set([special1])).toEqual(new Set([special2]));
+    expect(toIterator([special1, special2])).toEqual(
+      toIterator([special2, special1]),
+    );
+    expect([special1]).toContainEqual(special2);
+    expect({a: special1}).toHaveProperty('a', special2);
+    expect({a: special1, b: undefined}).toStrictEqual({
+      a: special2,
+      b: undefined,
+    });
+    expect({a: 1, b: {c: special1}}).toMatchObject({
+      a: 1,
+      b: {c: special2},
+    });
+  });
+
+  it('asymmetric matchers pass different special objects', () => {
+    expect([special1]).toEqual(expect.arrayContaining([special2]));
+    expect({a: 1, b: {c: special1}}).toEqual(
+      expect.objectContaining({b: {c: special2}}),
+    );
+  });
+
+  it('spy matchers pass different special objects', () => {
+    const mockFn: (...args: Array<any>) => any = jest.fn(() => specialReturn1);
```

Comment:
Would be better to use type argument and `unknown`. Somehow like this:

```suggestion
    const mockFn = jest.fn<(...args: Array<unknown>) => unknown>(() => specialReturn1);
```

Reply:
Oh nice! Updated to match.